addons/account/models/partner.py

```diff
@@ -433,7 +433,15 @@ def _get_company_currency(self):
                 partner.currency_id = self.env.company.currency_id
 
     credit = fields.Monetary(compute='_credit_debit_get', search=_credit_search,
-        string='Total Receivable', help="Total amount this customer owes you.")
+        string='Total Receivable', help="Total amount this customer owes you.",
+        groups='account.group_account_invoice,account.group_account_readonly')
+    credit_limit = fields.Monetary(string='Internal Credit Limit', default=-1)
```

Comment:
If the feature is enabled/disabled by company, why is the credit_limit the same for all the companies ?

What if one company wants to apply the company default and another one wants to specify a given limit ?

Reply:
This is a good point, thanks for raising it :slightly_smiling_face: 
I've discussed it with our POs, and it seems like premature optimization at this point.
It has advantages and disadvantages.
If you want to specify a limit for a partner (high or low) because you've evaluated its risk profile, it makes sense to use it across all the companies.
Its risk profile will not change depending on which company is doing the sale/invoicing.
And it would be a bit of a hassle to have to specify the same limit on the same partner multiple times if there are several companies.
So, the idea is to wait and see if we get feedback asking for it, and then implement it :wink:

Reply:
Another good point :slightly_smiling_face: 
It is indeed a major concern that would break the whole feature in a multi-currency environment.
So we had to revise the spec in order to make the partner credit limit company dependent, as you suggested, even though we wanted to avoid it :wink: 
But it's better to be bothered by having to set the limit on the same partner multiple times, rather than having inconsistent values across currencies.
Thanks again for the heads up!